analyses/tumor-gtex-plots/README.md

```diff
@@ -0,0 +1,124 @@
+## Expression Plots
+
+**Module authors:** Komal Rathi ([@komalsrathi](https://github.com/komalsrathi))
+
+### Purpose
+
+1. Create `tumor vs normal` and `tumors only` expression plots, with `TPM` on the y-axis, as follows:
+
+* `cohort + cancer_group tumor only plots`: For each cohort + cancer_group, plot all tumors.
+* `cancer_group tumor only plots`: For each cancer_group, plot all tumors.
+* `cohort + cancer_group tumor vs normal plots`: For each cancer_group within a cohort, plot cohort + cancer_group vs GTEx subgroups.
+* `cancer_group tumor vs normal plots`: For each cancer_group, plot cancer_group vs GTEx subgroups.
+
+2. Export two tables (.tsv) `cancer_group_level.tsv` and `cohort_cancer_group_level.tsv` with the following fields: `gene`, `ENSG_id`, `cohort`, `cancer_group`, `x_labels`, `mean`, `median`, `sd`, `efo_code`, `mondo_code`, `uberon_code`, `plot_api`.
+
+3. Export metadata files to use in JSON files with the following fields: `gene` (gene name), `plot_type` (either `tumors_only` or `tumor_vs_normal`), `cohort` (`_` separated cohort name), `cancer_group` (only applicable for tumor vs normal plots and `NA` for tumors only plot), `analysis_type` (either `cohort_cancer_group_level` or `cancer_group_level`), `plot_fname` (plot filename) and `table_fname` (table filename).
+
+### Methods 
+
+1. Filter `cancer_subgroup` or `gtex_subgroup` with number of samples < 5 
+2. Currently only generating outputs for `GPC2` and `MYCN` for the initial review
+
+### Module structure
+
+```
+├── 01-tumor-gtex-plots.R # script to call all functions and run analysis
+├── README.md 
+├── plots # png files of expression boxplots (possible examples)
+│   ├── GPC2_GMKF_Neuroblastoma_vs_GTEx_cohort_cancer_group_level.png
+│   ├── GPC2_PBTA_Medulloblastoma_vs_GTEx_cohort_cancer_group_level.png
+│   ├── GPC2_Ependymoma_vs_GTEx_cancer_group_level.png
+│   ├── GPC2_PBTA_GMKF_pan_cancer_cancer_group_level.png
+│   ├── GPC2_PBTA_GMKF_pan_cancer_cohort_cancer_group_level.png
+│   └── ...
+├── results
+│   ├── cancer_group_level.tsv 
+│   ├── cohort_cancer_group_level.tsv
+│   └── metadata.tsv # mapping of filename and metadata for all comparisons 
+├── run-tumor-gtex-plots.sh # full analysis script
+├── run-tumor-only-plots.sh # script for tumor only plots
+├── run-tumor-vs-normal-plots.sh # script for tumor vs gtex plots
+└── util
+    ├── pubTheme.R # publication quality ggplot2 theme
+    ├── tumor_plot.R # function for tumor only plot
+    └── tumor_vs_normal_plot.R # function for tumor vs normal plot
+```
+
+### Analysis scripts
+
+#### 01-tumor-gtex-plots.R
+
+##### Input parameters:
+
+```
+Rscript 01-tumor-gtex-plots.R --help
+
+Options:
+	--expr_mat=EXPR_MAT
+		collapsed TPM expression data: HUGO gene symbol x Sample identifiers (.rds)
+
+	--hist_file=HIST_FILE
+		histologies file (.tsv)
+
+	--cohort_list=COHORT_LIST
+		comma separated list of cohorts
```

Comment:
```suggestion
	--cohort_list=COHORT_LIST
		comma separated list of cohorts
```

For this, we are going to use `all_cohorts` if we have >1 cohort, so can you update the code and output table to reflect that?

Reply:
Actually I kept it this way so that we can use either `--cohort_list 'GMKF, PBTA, GTEx'` for tumor vs normal plots and `--cohort_list 'GMKF, PBTA'` for tumors only plots. Because for tumors only plots, we are not really using all cohorts.